src/main/java/com/minecolonies/coremod/entity/pathfinding/Pathfinding.java

```diff
@@ -91,153 +90,156 @@ public static void debugDraw(final double frame)
+        final VertexBuffer vertexBuffer = tessellator.getBuffer();
+        vertexBuffer.begin(GL11.GL_QUADS, DefaultVertexFormats.POSITION);
+        GlStateManager.color(r, g, b);
+
+        //  X+
+        vertexBuffer.pos(1.0, 0.0, 0.0).endVertex();
+        vertexBuffer.pos(1.0, 1.0, 0.0).endVertex();
+        vertexBuffer.pos(1.0, 1.0, 1.0).endVertex();
+        vertexBuffer.pos(1.0, 0.0, 1.0).endVertex();
+
+        //  X-
+        vertexBuffer.pos(0.0, 0.0, 1.0).endVertex();
+        vertexBuffer.pos(0.0, 1.0, 1.0).endVertex();
+        vertexBuffer.pos(0.0, 1.0, 0.0).endVertex();
+        vertexBuffer.pos(0.0, 0.0, 0.0).endVertex();
 
         //  Z-
-        GL11.glVertex3d(0.0, 0.0, 0.0);
-        GL11.glVertex3d(0.0, 1.0, 0.0);
-        GL11.glVertex3d(1.0, 1.0, 0.0);
-        GL11.glVertex3d(1.0, 0.0, 0.0);
+        vertexBuffer.pos(0.0, 0.0, 0.0).endVertex();
+        vertexBuffer.pos(0.0, 1.0, 0.0).endVertex();
+        vertexBuffer.pos(1.0, 1.0, 0.0).endVertex();
+        vertexBuffer.pos(1.0, 0.0, 0.0).endVertex();
 
         //  Z+
-        GL11.glVertex3d(1.0, 0.0, 1.0);
-        GL11.glVertex3d(1.0, 1.0, 1.0);
-        GL11.glVertex3d(0.0, 1.0, 1.0);
-        GL11.glVertex3d(0.0, 0.0, 1.0);
+        vertexBuffer.pos(1.0, 0.0, 1.0).endVertex();
+        vertexBuffer.pos(1.0, 1.0, 1.0).endVertex();
+        vertexBuffer.pos(0.0, 1.0, 1.0).endVertex();
+        vertexBuffer.pos(0.0, 0.0, 1.0).endVertex();
 
         //  Y+
-        GL11.glVertex3d(1.0, 1.0, 1.0);
-        GL11.glVertex3d(1.0, 1.0, 0.0);
-        GL11.glVertex3d(0.0, 1.0, 0.0);
-        GL11.glVertex3d(0.0, 1.0, 1.0);
+        vertexBuffer.pos(1.0, 1.0, 1.0).endVertex();
+        vertexBuffer.pos(1.0, 1.0, 0.0).endVertex();
+        vertexBuffer.pos(0.0, 1.0, 0.0).endVertex();
+        vertexBuffer.pos(0.0, 1.0, 1.0).endVertex();
 
         //  Y-
-        GL11.glVertex3d(0.0, 0.0, 1.0);
-        GL11.glVertex3d(0.0, 0.0, 0.0);
-        GL11.glVertex3d(1.0, 0.0, 0.0);
-        GL11.glVertex3d(1.0, 0.0, 1.0);
+        vertexBuffer.pos(0.0, 0.0, 1.0).endVertex();
+        vertexBuffer.pos(0.0, 0.0, 0.0).endVertex();
+        vertexBuffer.pos(1.0, 0.0, 0.0).endVertex();
+        vertexBuffer.pos(1.0, 0.0, 1.0).endVertex();
 
-        GL11.glEnd();
+        tessellator.draw();
 
         if (n.parent != null)
         {
-            GL11.glBegin(GL11.GL_LINES);
-            GL11.glColor3f(0.75F, 0.75F, 0.75F);
-
             final double pdx = n.parent.pos.getX() - n.pos.getX() + 0.125;
             final double pdy = n.parent.pos.getY() - n.pos.getY() + 0.125;
             final double pdz = n.parent.pos.getZ() - n.pos.getZ() + 0.125;
 
-            GL11.glVertex3d(0.5, 0.5, 0.5);
-            GL11.glVertex3d(pdx / 0.25, pdy / 0.25, pdz / 0.25);
-
-            GL11.glEnd();
+            vertexBuffer.begin(GL11.GL_LINES, DefaultVertexFormats.POSITION_COLOR);
+            vertexBuffer.pos(0.5D, 0.5D, 0.5D).color(0.75F, 0.75F, 0.75F, 1.0F).endVertex();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 0.5D to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 0.5D to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 0.5D to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 0.75F to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 0.75F to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 0.75F to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)